files/image_config/rsyslog/rsyslog.conf.j2

```diff
@@ -107,19 +107,24 @@ $RepeatedMsgReduction on
 {% set regex = conf.get('filter_regex') -%}
 
 {% set fmodifier = '!' if filter == 'exclude' else '' %}
-{% set device = 'eth0' if vrf == 'default' else vrf -%}
+{% set device = vrf if vrf != '' and vrf != 'default' -%}
```

Comment:
>if [](http://example.com/codeflow?start=20&length=2)

There is no `else` after this `if`. Will it break?

Reply:
Hi
No.
If the condition `vrf != '' and vrf != 'default'` isn't satisfied, `device` is empty, which is expected.
In line [121](https://github.com/sonic-net/sonic-buildimage/pull/17616/files#diff-b5c3b035d28139ed77348578faf3d18d55884f61a98bb218fca31c993725a98eR121), the `device` is empty and option will be empty as well.
We do not pass `device` without `vrf` specified as a non-default one, which is the motivation of this PR.
This case is covered by the 1st manual test item `Configuring a syslog server without VRF specified or with default as the VRF: no device passed in rsyslog.conf`